client/src/components/AskQuestionBoard.jsx

```diff
@@ -33,6 +54,58 @@ class AskQuestionBoard extends React.Component {
     });
   }
 
+  handleQuestionDialogClose() {
+    this.setState({
+      questionDialog: false,
+      question: '',
+      errorText: '',
+    });
+  }
+
+  handleQuestionDialogSubmit() {
+    if ( this.state.question === '' ) {
+      this.setState({
+        errorText: 'Question cannot be empty'
+      });
+    } else {
+      this.setState({
+        questionDialog: false,
+      });
+      this.addQuestion(this.state.user, this.state.question);
+      this.setState({
+        question: '',
+        errorText: '',
+      });
+    }
+  }
+
+  handleQuestionChange(event) {
+    this.setState({
+      question: event.target.value,
+    });
+  }
+
+  handleAnswerChange(event) {
+    this.setState({
+      answer: event.target.value,
+    });
+  }
+
+  handleAnswerSubmit(event) {
+    if (event.charCode === 13) {
```

Comment:
Its generally better to not have hardcoded integers in your code. I would make this a constant instead.